utils.go

```diff
@@ -0,0 +1,64 @@
+package main
+
+import (
+	"net"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// Returns the preferedIPProtocol, the dialProtocol, and sets the probeIPProtocolGauge.
+func configureProtocol(preferredIPProtocol string, targetAddress string, module Module, probeIPProtocolGauge prometheus.Gauge) (string, string, error) {
```

Comment:
You don't need to know the module

Also, create the Gauge in this function. Take in the registry.